doc/source/overview/config_files.md

```diff
@@ -0,0 +1,753 @@
+:widths: auto
+
+* - Field
+  - Description
+  - Default (if optional)
+* - `kernel_a`
+  -  Kernel to apply to projected inputs. A string referencing a kernel serialized in a `.dill` file, a registered 
+  kernel object, or a kernel artefact dictionary (see {ref}`kernel-table`).
+  - `"@utils.tensorflow.kernels.GaussianRBF"` <br/>(with `trainable = True`)
+* - `kernel_b`
+  -  Kernel to apply to raw inputs. A string referencing a kernel serialized in a `.dill` file, a registered kernel 
+  object, or a kernel artefact dictionary (see {ref}`kernel-table`).
+  - `None`
+* - `proj`
+  - Projection to be applied to the inputs. Should be a TensorFlow model specified as a registered 
+  model, or as an artefact dictionary following the layout in {ref}`model-table`. 
+  - 
+* - `eps`
+  - The proportion (in [0,1]) of weight to assign to `kernel_b`. Specified as a float, or set to `"trainable"`. 
+  - `"trainable"`
+```
+
+As shown in the example below, when a `DeepKernel` is specified, `kernel_a` and `kernel_b` are themselves kernels, 
+and can be specified via the {ref}`kernel-table`.
+
+<p class="codeblock-label">config.toml (excerpt)</p>
+
+```toml
+[kernel]
+eps = 0.01
+
+[kernel.kernel_a]
+src = "@utils.tensorflow.kernels.GaussianRBF"
+trainable = true
+
+[kernel.kernel_b]
+src = "custom_kernel.dill"
+sigma = [ 1.2,]
+trainable = false
+
+[kernel.proj]
+src = "model/"
+```
+
+#### Optimizers
+
+Optimizers, required by detectors such as `LearnedKernelDrift` and `ClassifierDrift`, are specified via the
+`optimizer` field. For the `tensorflow` backend, `optimizer` should be specified as an artefact dictionary following the
+config schema expected by 
+[tf.keras.optimizers.deserialize](https://www.tensorflow.org/api_docs/python/tf/keras/optimizers/deserialize).
+A TensorFlow `Optimizer` config can be generated with the 
+[tf.keras.optimizers.serialize](https://www.tensorflow.org/api_docs/python/tf/keras/optimizers/serialize) function.
+
+Example:
+
+<p class="codeblock-label">config.toml (excerpt)</p>
+
+```toml
+[optimizer]
+class_name = "Adam"
+
+[optimizer.config]
+name = "Adam"
+learning_rate = 0.01
+beta_1 = 0.89
+beta_2 = 0.99
+epsilon = 1e-7
+amsgrad = false
+```
+
+(registering_artefacts)=
+### Registering artefacts
+
+Custom artefacts defined in Python code may be specified in the config file without the need to serialise them, 
+by first adding them to the Alibi Detect artifact registry using the [registry](../api/alibi_detect.utils.registry.rst) 
+submodule. This submodule harnesses the [catalogue](https://github.com/explosion/catalogue) library to allow functions 
+to be registered with a decorator syntax:
+
+
+`````{grid} 2
+
+````{grid-item-card}
+:shadow: md
+:margin: 1
+:padding: 1
+:columns: auto
+
+**Registering a function**
+^^^
+
+```python
+import numpy as np
+from alibi_detect.utils.registry import registry
+from alibi_detect.utils.loading import load_detector
+
+# Register a simple function
+@registry.register('my_function.v1')
```

Comment:
Slightly off-topic, what happens if no name is passed to the function? I would think a nice default would be to automatically register with the function name, but not sure if `catalogue` does it. If not then no worries.

Reply:
OK we can open an issue afterwards.